Doc/reference/expressions.rst

```diff
@@ -1380,7 +1405,9 @@ in mathematics:
    comp_operator: "<" | ">" | "==" | ">=" | "<=" | "!="
                 : | "is" ["not"] | ["not"] "in"
 
-Comparisons yield boolean values: ``True`` or ``False``.
+Comparisons yield boolean values: ``True`` or ``False``. Custom
+:dfn:`rich comparison methods` may return non-boolean values. In this case
+Python will call :func:`bool` on such value to convert it to boolean as needed.
```

Comment:
@ambv on a 2nd thought, the problem is this phrasing can be misleading:

[ins] In [7]: class A:
         ...:     def __lt__(s, o):
         ...:         return 'ay'
         ...:
         ...:

[ins] In [8]: A() < 5
Out[8]: 'ay'

Current text can be read to imply bool() would be called before this value is printed or used somewhere, it's not obvious IMO that 'as needed' refers only to `if` constructs.

Reply:
"as needed", e.g. when a boolean context is expected. But that gets too wordy to spell out and repeats the word "boolean" like three times.

Reply:
What do you think about "Python will call :func:`bool` on such value in boolean contexts." ?

Reply:
Sounds good, I like it.